lib/compiler-cranelift/src/translator/code_translator/bounds_checks.rs

```diff
@@ -0,0 +1,387 @@
+//! Implementation of Wasm to CLIF memory access translation.
+//!
+//! Given
+//!
+//! * a dynamic Wasm memory index operand,
+//! * a static offset immediate, and
+//! * a static access size,
+//!
+//! bounds check the memory access and translate it into a native memory access.
+//!
+//! !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Comment:
As above, is this copied?

Reply:
Yup, is copied but already attributed:
https://github.com/wasmerio/wasmer/tree/6df1e38bd9af4fba2219cf1e54aba975f0f39b77/lib/compiler-cranelift#acknowledgments